src/navigation/_AppSettingsProvider.jsx

```diff
@@ -0,0 +1,82 @@
+import React, {
+  useState,
+  useEffect,
+  useMemo,
+} from 'react';
+import PropTypes from 'prop-types';
+import AppSettingsContext from './_AppSettingsContext';
+import { settingsConfigPropType } from '../site/siteConfigPropTypes';
+
+const propTypes = {
+  children: PropTypes.element.isRequired,
+  settingsConfig: settingsConfigPropType.isRequired,
+};
+
+const AppSettingsProvider = ({ settingsConfig, children }) => {
+  const {
+    defaultLocale = 'en',
+    defaultTheme,
+    defaultDirection = 'ltr',
+    themes,
+  } = settingsConfig;
+
+  const [currentLocale, setCurrentLocale] = useState(defaultLocale);
+  const [currentDirection, setCurrentDirection] = useState(defaultDirection);
+  const [currentTheme, setCurrentTheme] = useState(defaultTheme);
+
+  /**
+   * Place settings on dom
+   */
+  useEffect(() => {
+    const htmlNode = document.getElementsByTagName('html')[0];
+
+    if (htmlNode.getAttribute('lang') !== currentLocale) {
+      htmlNode.setAttribute('lang', currentLocale);
+    }
+
+    if (htmlNode.getAttribute('dir') !== currentDirection) {
+      htmlNode.setAttribute('dir', currentDirection);
+    }
+  }, [currentLocale, currentDirection]);
+
+  const AppSettings = useMemo(() => {
```

Comment:
Nitpicking here, but it's weird to capitalize this.

Reply:
Yep, lowercased